feat: Option to automatically create share links for recipes in recipe actions

ianp1 opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.

  To start, try providing a short summary of your changes in the Title above. We follow Conventional Commits syntax, please ensure your title is prefixed with one of:
  - `feat: `
  - `fix: `
  - `docs: `
  - `chore: `
  - `dev:`

  If a section of the PR template does not apply to this PR, then delete that section.

  PLEASE READ:
  -------------------------
  Mealie is moving to a regular, automatic release schedule. This means that all PRs should be in a
  stable state, ready for release. This includes:

  - Ensuring new tests have been added to cover new features, or to prevent regressions.
  - Work is fully complete and usable

 -->

## What type of PR is this?


<!--
  Delete any of the following that do not apply:
 -->

- feature
- bug

## What this PR does / why we need it:


<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.

  If there is a UI component to the change, please include before/after images.
-->
Currently, when using a recipe action on a recipe that is not publicly available (due to recipe, group or household being private), the receiving service will not be able to read the recipe data. 

As discussed in #4619, this PR implements an additional template option for recipe actions. The option `${share-link-expires-seconds-<x>}` is automatically replaced by a newly created share link that expires after `x` seconds. This makes share links usable even on private recipes, groups and households. 


## Which issue(s) this PR fixes:


<!--
If this PR fixes one of more issues, list them here.
One per line, like so:
Fixes #123
Fixes #39
-->
Fixes  #4619

## Special notes for your reviewer:
I am not completely certain, if the method to obtain the group slug is correct. Please check this part thoroughly.
Also, i have not yet created documentation for the new option - is the website documentation part of this repository?

<!--
   Is there any particular feedback you would / wouldn't like?
   Which parts of the code should reviewers focus on?
-->

## Testing
I created a private recipe, a public recipe and multipe recipe actions

- One containing a single share link template
- One containing multiple share link templates with different expiry times
- And one containing multiple share link templates, some of which having the same expiry times

I also tested with different privacy settings in group and household. The receiver of the given action was always able to read the recipes, without being logged in using the share link.


<!--
  Describe how you tested this change.
-->

ianp1 committed:
add expiring share links to url recipe actions

ianp1 committed:
add expiring share links to url recipe actions

michael-genson committed:
Merge branch 'mealie-next' into bug/private-recipe-recipe-actions

michael-genson commented on frontend/composables/use-group-recipe-actions.ts:
```diff
@@ -46,8 +45,40 @@ export const useGroupRecipeActions = function (
+  async function parseRecipeActionUrl(url: string, recipe: Recipe): Promise<string> {
     /* eslint-disable no-template-curly-in-string */
+    const shareLinkRegex = /\$\{share-link-expires-seconds-[0-9]+\}/g;
+    const group = (await api.groups.getOne(recipe.groupId || "")).data;
```
You should be able to grab the slug from the URL instead of fetching it via the API:

```js
import { useRoute } from "@nuxtjs/composition-api";

// put this right under "useUserApi"
const route = useRoute();

// replace the API call with this
const groupSlug = route.value.params.groupSlug || recipe.groupId || "";
```

michael-genson reviewed (changes requested):
Looks great! Sorry for taking so long on this, holidays beat me up. One change to remove an API call and we're all set.

Would you also be able to add this to the docs? Just add it to the list of merge fields and a short (one sentence) description. And then also maybe change the Bring example to use the new field (instead of `url`).
https://docs.mealie.io/documentation/getting-started/features/#recipe-actions

renovate[bot] committed:
fix(deps): update dependency alembic to v1.14.1 (#4915)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency orjson to v3.10.15 (#4911)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
fix: Show All Recipes in Cookbook Regardless of Sort (#4908)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#4917)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.59.9 (#4919)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ChipSkylark37 committed:
docs: Add to FAQ: When / how to use Linked Ingredients (#4897)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pre-commit to v4.1.0 (#4922)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Commit:
docs: Add community guide for integrating the Mealie-Bring-API (#4920)

hay-kot committed:
chore(l10n): New Crowdin updates (#4926)

Kuchenpirat committed:
fix: prevent recipe sharing from different group (#4929)

Kuchenpirat committed:
fix: prevent users from updating their own household privileges (#4928)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.60.0 (#4932)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4934)

github-actions[bot] committed:
docs(auto): Update image tag, for release v2.5.0 (#4933)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.7 (#4935)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4938)

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.3 (#4939)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.60.1 (#4941)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

james-d-elliott committed:
docs: update oidcv2 authelia link (#4942)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

ianp1 commented:
Thanks, good to hear! I am unfortunately unavailable till the start of February, but I will have a look at it then

Greetings

Wetzel402 committed:
feat: Shopping list UI overhaul - three dot menu (#4415)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4953)

michael-genson committed:
feat: Add Servings/Yield to Recipe Actions (#4952)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#4965)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.60.2 (#4967)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4968)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4975)

renovate[bot] committed:
chore(deps): update dependency pylint to v3.3.4 (#4970)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pytest-asyncio to v0.25.3 (#4972)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

felixschndr committed:
fix: Ensure bring api docs are shown (PR 4920) (#4948)

Co-authored-by: Hayden <64056131+hay-kot@users.noreply.github.com>

michael-genson committed:
fix: Remove API Tokens from User APIs (#4985)

PancakeZik committed:
fix: Fixed LastMade recipes sorting order (#4980)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.4 (#4988)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency authlib to v1.4.1 (#4971)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.8 (#4989)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#4991)

renovate[bot] committed:
fix(deps): update dependency rapidfuzz to v3.12.1 (#4990)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Cameronwyatt committed:
feat: Add new labels and foods for en-US language

Update all locale seeding files and seeding logic to parse the new format Only add new labels, units, and foods during seeding (checking against existing names)

hay-kot committed:
fix: revert "feat: Add new labels and foods for en-US language" (#4996)

renovate[bot] committed:
fix(deps): update dependency openai to v1.61.0 (#4995)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.1 (#4992)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5008)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.2 (#5011)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5016)

hay-kot committed:
chore(l10n): New Crowdin updates (#5018)

renovate[bot] committed:
chore(deps): update dependency mypy to v1.15.0 (#5017)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5021)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.61.1 (#5020)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency beautifulsoup4 to v4.13.3 (#5009)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs(auto): Update image tag, for release v2.6.0 (#5022)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5032)

renovate[bot] committed:
fix(deps): update dependency pydantic to v2.10.6 (#4940)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
chore: Remove Warnings (#5039)

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.3 (#5031)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.5 (#5029)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency sqlalchemy to v2.0.38 (#5030)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5046)

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5047)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

chishm committed:
dev: Create a Python package, build Docker images from it (#4551)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>
Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>
Co-authored-by: Hayden <64056131+hay-kot@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.6 (#5049)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
fix: shorten indexes to fix issues with index limits (#5045)

RMI78 committed:
feat: support `_FILE` suffix for docker secrets (again) (#4958)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency coverage to v7.6.12 (#5042)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
fix(deps): update dependency openai to v1.63.0 (#5067)

renovate[bot] committed:
fix(deps): update dependency openai to v1.63.2 (#5070)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency lxml to v5.3.1 (#5048)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.4 (#5051)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5073)

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.5 (#5078)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5080)

renovate[bot] committed:
fix(deps): update dependency recipe-scrapers to v15.5.0 (#5087)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency recipe-scrapers to v15.5.1 (#5089)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pydantic-settings to v2.8.0 (#5086)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5093)

renovate[bot] committed:
fix(deps): update dependency openai to v1.64.0 (#5092)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
fix: Only run migration data fixes on migrations (#5038)

miah120 committed:
feat: Improve Shopping List UI (#4608)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.7 (#5079)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5069)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5098)

hay-kot committed:
chore(l10n): New Crowdin updates (#5099)

hay-kot committed:
chore(l10n): New Crowdin updates (#5100)

hay-kot committed:
chore(l10n): New Crowdin updates (#5101)

hay-kot committed:
chore(l10n): New Crowdin updates (#5102)

michael-genson committed:
feat: Better Scraping/More User Agents (#5091)

renovate[bot] committed:
fix(deps): update dependency beautifulsoup4 to v4.13.3 (#5090)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ChipSkylark37 committed:
docs:  Add additional information and tips to Backup & Restore Usage Documentation Page & Shopping List (#4843)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

michael-genson committed:
fix: Invalidate Expired Shared Links (#5065)

renovate[bot] committed:
fix(deps): update dependency authlib to v1.5.0 (#5103)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Kuchenpirat committed:
feat: redesign recipe info card (#5026)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5105)

hay-kot committed:
chore(l10n): New Crowdin updates (#5106)

Kuchenpirat committed:
fix: update recipe time row direction on small screens (#5107)

michael-genson committed:
fix: Bulk Add Recipes to Shopping List (#5054)

renovate[bot] committed:
fix(deps): update dependency pydantic-settings to v2.8.1 (#5108)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs(auto): Update image tag, for release v2.7.0 (#5111)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.65.1 (#5123)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.8 (#5112)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.9 (#5122)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

michael-genson committed:
fix: Remove `br` encoding from scraper (#5115)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.9 (#5130)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs(auto): Update image tag, for release v2.7.1 (#5129)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency bcrypt to v4.3.0 (#5127)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5131)

michael-genson committed:
feat: Migrate from CRF++ to Ingredient Parser (a Python package) (#5061)

michael-genson committed:
chore: Renovate Ignore Python Upgrades (#5134)

hay-kot committed:
chore(l10n): New Crowdin updates (#5135)

michael-genson committed:
fix(deps): update dependency ingredient-parser-nlp to v2 (#5137)

renovate[bot] committed:
fix(deps): update dependency authlib to v1.5.1 (#5138)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5142)

hay-kot committed:
chore(l10n): New Crowdin updates (#5145)

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.10 (#5139)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.6 (#5141)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.65.2 (#5147)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.11 (#5150)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5149)

renovate[bot] committed:
fix(deps): update dependency rapidfuzz to v3.12.2 (#5159)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pytest to v8.3.5 (#5158)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

eric-hoffmann committed:
dev: chown commandhistory to correct user during devcontainer creation (#5166)

hay-kot committed:
chore(l10n): New Crowdin updates (#5161)

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.7 (#5163)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5167)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>

eric-hoffmann committed:
fix: Use recipe-parsers nutrients function for nutrition parsing (#5165)

hay-kot committed:
chore(l10n): New Crowdin updates (#5174)

hay-kot committed:
chore(l10n): New Crowdin updates (#5175)

hay-kot committed:
chore(l10n): New Crowdin updates (#5176)

renovate[bot] committed:
fix(deps): update dependency openai to v1.65.3 (#5180)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency jinja2 to v3.1.6 [security] (#5183)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.65.4 (#5182)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5179)

michael-genson committed:
fix: Filter out null chars from OpenAI response (#5187)

renovate[bot] committed:
chore(deps): update dependency ruff to v0.9.10 (#5188)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency recipe-scrapers to v15.6.0 (#5198)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

brycied00d committed:
docs: Re-add missing environment variable documentation (#5199)

renovate[bot] committed:
chore(deps): update dependency pylint to v3.3.5 (#5194)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5200)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

michael-genson committed:
fix: Case Insensitive Query Filters (#5162)

myxor committed:
docs: faq.md: fix paths to scripts (#5201)

Cameronwyatt committed:
feat: Update seeding to use new foods list format - round 2 (#5189)

hay-kot committed:
fix: Revert "feat: Update seeding to use new foods list format - round 2" (#5208)

TeaRex-coder committed:
fix: PostgreSQL capitalization (#5220)

cmintey committed:
feat: OIDC: Call userinfo if no claims found in id token (#5228)

Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.9 (#5209)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5229)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

michael-genson committed:
chore: Bump Ruff to 0.11.0 (#5233)

hay-kot committed:
chore(l10n): New Crowdin updates (#5185)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

michael-genson committed:
fix: "NOT IN" doesn't apply filter properly (#5154)

renovate[bot] committed:
fix(deps): update dependency alembic to v1.15.1 (#5178)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.66.5 (#5197)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency sqlalchemy to v2.0.39 (#5204)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pillow-heif to ^0.22.0 (#5219)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency coverage to v7.7.0 (#5227)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
docs(auto): Update image tag, for release v2.8.0 (#5236)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pre-commit to v4.2.0 (#5238)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Kuchenpirat committed:
fix: spelling of github (#5244)

hay-kot committed:
chore(l10n): New Crowdin updates (#5243)

hay-kot committed:
chore(l10n): New Crowdin updates (#5245)

hay-kot committed:
chore(l10n): New Crowdin updates (#5246)

renovate[bot] committed:
fix(deps): update dependency openai to v1.67.0 (#5247)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5248)

hay-kot committed:
chore(l10n): New Crowdin updates (#5250)

renovate[bot] committed:
fix(deps): update dependency openai to v1.68.0 (#5254)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.1 (#5253)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.68.2 (#5259)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5252)

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.2 (#5258)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Kuchenpirat committed:
fix: update admin scripts paths in docs (#5263)

renovate[bot] committed:
fix(deps): update dependency fastapi to v0.115.12 (#5268)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5269)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>

Kuchenpirat committed:
fix: remove unmaintained ios shortcut (#5280)

hay-kot committed:
chore(l10n): New Crowdin updates (#5262)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

michael-genson committed:
fix: Pre-download NLTK during Docker build (#5290)

Kuchenpirat committed:
dev: add pull request image build workflow (#5235)

renovate[bot] committed:
chore(deps): update dependency pylint to v3.3.6 (#5251)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency coverage to v7.7.1 (#5260)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency orjson to v3.10.16 (#5270)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pytest-asyncio to ^0.26.0 (#5274)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency python-dotenv to v1.1.0 (#5275)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency typing-extensions to v4.13.0 (#5278)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency sqlalchemy to v2.0.40 (#5283)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.69.0 (#5284)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pydantic to v2.11.1 (#5285)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency alembic to v1.15.2 (#5289)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.10 (#5293)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency rich to v14 (#5294)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency apprise to v1.9.3 (#5295)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency coverage to v7.8.0 (#5297)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5281)

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.11 (#5304)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5301)

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.3 (#5311)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pydantic to v2.11.2 (#5312)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.70.0 (#5300)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency authlib to v1.5.2 (#5308)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency rapidfuzz to v3.13.0 (#5314)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.4 (#5317)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency lxml to v5.3.2 (#5318)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5320)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.71.0 (#5322)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Kuchenpirat committed:
fix: build pull request image only in mealie repo (#5327)

renovate[bot] committed:
fix(deps): update dependency openai to v1.72.0 (#5328)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5310)

Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pydantic to v2.11.3 (#5325)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.73.0 (#5335)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.5 (#5333)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency typing-extensions to v4.13.2 (#5313)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5344)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency pillow to v11.2.1 (#5337)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

aclerici38 committed:
docs: Add community docs for an iOS shortcut (attempt 2) (#5345)

renovate[bot] committed:
fix(deps): update dependency openai to v1.74.0 (#5346)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency openai to v1.75.0 (#5357)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5360)

renovate[bot] committed:
chore(deps): update dependency mkdocs-material to v9.6.12 (#5359)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency ruff to v0.11.6 (#5361)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency beautifulsoup4 to v4.13.4 (#5352)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency aniso8601 to v10.0.1 (#5368)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency html2text to v2025 (#5347)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5370)

github-actions[bot] committed:
chore(auto): Update pre-commit hooks (#5372)

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>
Co-authored-by: Kuchenpirat <24235032+Kuchenpirat@users.noreply.github.com>

hay-kot committed:
chore(l10n): New Crowdin updates (#5374)

Abalysa commented:
Is there any follow up on this PR?
I am facing the same issue that (even tough my mealie instance itself is accessible to the internet) i dont want to mark all my recipes as public just for the Bring! integration.
@ianp1, @michael-genson

hay-kot committed:
chore(l10n): New Crowdin updates (#5379)

renovate[bot] committed:
fix(deps): update dependency openai to v1.76.0 (#5381)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency uvicorn to v0.34.2 (#5343)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

robertdanahome committed:
fix: Add missing group_id to RecipeTag and TagBase schemas (#5342)

Co-authored-by: Robert Dana <bob@yall.org>
Co-authored-by: Michael Genson <71845777+michael-genson@users.noreply.github.com>

ianp1 committed:
docs: add documentation for "share-link-expires-seconds" recipe action (#4676)

ianp1 commented:
Hey there,

i am sorry, live has been busy and i have forgotten about this pull request. I have now added the requested documentation. @michael-genson is everything as expected?

I have also pulled my branch up to current state of mealie-next and tested everything again. However, rebasing seems to have messed up this merge request commits - should i try to recreate this PR with a clean branch?

Greetings

michael-genson commented:
Something seems up with the recent commits, it doesn't look like the main branch was merged in.
Apologies for the delay, the maintainers (myself included) have been quite busy, and frontend PRs are currently blocked by [the Nuxt 3 upgrade](https://github.com/mealie-recipes/mealie/pull/5184).

ianp1 commented:
Do you have any advice on how to proceed? I would create a new PR with cleaned up history, or do you have a better idea?

michael-genson commented:
Either that, or you can revert the erroneous commits and make sure you merge mealie-next into your branch

michael-genson commented:
I tried force-pushing a fixed commit history but that didn't seem to work. My recommendation would be to close this PR/branch and create a new branch with your changes. It will also help avoid the merge conflicts with the Nuxt upgrade

michael-genson commented:
Hey there, this PR has been sitting for a while and has some merge conflicts to be resolved. Since it's been a bit, I'm going to close this PR. Feel free to create a new PR, or you can re-open this one if you're able to resolve the conflicts.
